linode/linode-cli
# new: Improve generated command help page formatting using Rich

## 📝 Description

This change makes various large formatting improvements to command help pages, including:

- Bolding help page sections
- Adding color/bolding to argument names and prefixes
- Grouping arguments by parent and whether they are required
- Applying indentation to child arguments
- Converting Markdown links to human-readable absolute links

Additionally, this PR moves help page helpers from `arg_helpers.py` to a new `help_pages.py` file in an effort to keep files tightly scoped

## ✔️ How to Test

The following test steps assume you have pulled this code locally and run `make install`.

### Unit Testing

```
make testunit
```

### Integration Testing

```
make testint
```

### Manual Testing

1. View the help page for any CLI command (e.g. `linode-cli linodes ls --help`).
2. Ensure the output matches the criteria mentioned in the description and is readable/understandable.

## 📷 Preview



| linodes ls                                                                                                                                            | linodes create                                                                                                                                                        |
|-------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| <img width="300" alt="linode-cli linodes ls --help" src="https://github.com/linode/linode-cli/assets/114949949/dff0702e-8462-4ada-a190-58396f96046f"> | <img width="300" alt="linode-cli linodes create --help (arguments)" src="https://github.com/linode/linode-cli/assets/114949949/62864acc-9dd0-486a-bbfb-9648c71256ec"> |

| firewalls update                                                                                                                                            | firewalls delete                                                                                                                                            |
|-------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------|
| <img width="300" alt="linode-cli firewalls update --help" src="https://github.com/linode/linode-cli/assets/114949949/2cfe3f29-a3d7-43d4-a579-7cc2a6bdc086"> | <img width="300" alt="linode-cli firewalls delete --help" src="https://github.com/linode/linode-cli/assets/114949949/bc2b18e4-77dc-48ae-ae55-46a1a8cf6317"> |
